pkg/agent/sysadvisor/plugin/qosaware/resource/memory/advisor.go

```diff
@@ -68,32 +73,41 @@ func NewMemoryResourceAdvisor(conf *config.Configuration, extraConf interface{},
 		metaReader: metaCache,
 		metaServer: metaServer,
 		emitter:    emitter,
+		sendChan:   make(chan types.InternalMemoryCalculationResult),
 	}
 
-	initializers := headroompolicy.GetRegisteredInitializers()
+	headroomPolicyInitializers := headroompolicy.GetRegisteredInitializers()
 	for _, headroomPolicyName := range conf.MemoryHeadroomPolicies {
-		initFunc, ok := initializers[headroomPolicyName]
+		initFunc, ok := headroomPolicyInitializers[headroomPolicyName]
 		if !ok {
 			klog.Errorf("failed to find registered initializer %v", headroomPolicyName)
 			continue
 		}
 		ra.headroomPolices = append(ra.headroomPolices, initFunc(conf, extraConf, metaCache, metaServer, emitter))
 	}
 
+	memoryAdvisorPluginInitializers := memadvisorplugin.GetRegisteredInitializers()
+	for _, memadvisorPluginName := range conf.MemoryAdvisorPlugins {
+		initFunc, ok := memoryAdvisorPluginInitializers[memadvisorPluginName]
+		if !ok {
+			klog.Errorf("failed to find registered initializer %v", memadvisorPluginName)
+			continue
+		}
+		ra.plugins = append(ra.plugins, initFunc(conf, extraConf, metaCache, metaServer, emitter))
+	}
+
 	return ra
 }
 
 func (ra *memoryResourceAdvisor) Run(ctx context.Context) {
 	period := ra.conf.SysAdvisorPluginsConfiguration.QoSAwarePluginConfiguration.SyncPeriod
 
-	go wait.Until(func() {
-		ra.update()
-	}, period, ctx.Done())
+	go wait.Until(ra.update, period, ctx.Done())
+	go wait.Until(ra.sendAdvices, period, ctx.Done())
```

Comment:
It's better to make update and send advice synchronized, i.e. send immediately after update.